Code/encoding/Structs/TimeModel.cpp

```diff
@@ -0,0 +1,106 @@
+
+#include <Structs/TimeModel.h>
+#include <TiltedCore/Serialization.hpp>
+
+using TiltedPhoques::Serialization;
+
+const int cDayLengthArray[12] = {31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
+
+int TimeModel::GetNumerOfDaysByMonthIndex(int aIndex)
+{
+    if (aIndex < 12)
+    {
+        return cDayLengthArray[aIndex];
+    }
+
+    return 0;
+}
+
+void TimeModel::Update(uint64_t aDelta)
+{
+    float deltaSeconds = static_cast<float>(aDelta) / 1000.f;
+    Time += (deltaSeconds * (TimeScale * 0.00027777778f));
+
+    if (Time > 24.f)
+    {
+        int maxDays = GetNumerOfDaysByMonthIndex(Month);
+
+        while (Time > 24.f)
```

Comment:
I am sure we can do better than this loop!

Reply:
Sure. I've got a few changes I need to push, but I didn't actually write that, it's been there. I'll look into it when I'm up tomorrow.